packages/twenty-server/src/workspace/workspace-sync-metadata/reflective-metadata.factory.ts

```diff
@@ -28,58 +45,80 @@ export class MetadataParser {
       (field) => !isGatedAndNotEnabled(field, workspaceFeatureFlagsMap),
     );
 
+    if (objectMetadata?.dataSourceSchema) {
+      const dataSource = await this.dataSourceRepository.findOne({
```

Comment:
Maybe we can add a parameter Record<string, id> and fetch all datasources of the workspace earlier to avoid too many queries? (1 query vs n queries with 1 per objectMetadata that defines a custom schema in the decorator)

Reply:
It can be a possibility, but as we're in a seed command I don't think it's going to change a lot.
In terme of performance I think it's a win win, querying all the datasources each time, or just when it's needed (as it's not going to happen often because it's only for core relation).

Reply:
BTW I find it more readable that way, but it's only my opinion :)

Reply:
As migration can now handle different schema, I think it's better to keep it

Reply:
Mmm I see, because I don't see any objectMetadata where objectMetadata?.dataSourceSchema would not be null currently (which is also why I'm ok to merge it like this since there won't be any performance impact but was just wondering if it was still useful) @magrinj

Reply:
Yes you're right it's not used for now, maybe it's something we will need later ?
cc @charlesBochet, what do you think about that ? Am I removing it ? :)

Reply:
Yes, I think we can remove it :)

Reply:
I have removed the code! Merging it!